blocks/datasets/__init__.py

```diff
@@ -529,11 +541,66 @@ def get_data(self, request):
             self.cache[i] = cache[request:]
         return tuple(data)
 
+    def get_epoch_iterator(self, **kwargs):
+        self.cache = [[] for source in self.sources]
+        return super(CachedDataStream, self).get_epoch_iterator(**kwargs)
+
     def _cache(self):
         for cache, data in zip(self.cache, next(self.child_epoch_iterator)):
             cache.extend(data)
```

Comment:
I think we should have an attribute documentation for the `cache` attribute, since it can be used by other classes.